src/main/java/io/beanmapper/core/unproxy/UnproxyResultStore.java

```diff
@@ -0,0 +1,43 @@
+package io.beanmapper.core.unproxy;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Singleton responsible for storing the results of unproxy-results. Allows for fast and thread-safe retrieval of results.
+ */
+public final class UnproxyResultStore {
+
+    private static volatile UnproxyResultStore INSTANCE;
+
+    private final Map<Class<?>, Class<?>> unproxyResultClassStore;
+
+    private UnproxyResultStore() {
+        unproxyResultClassStore = Collections.synchronizedMap(new HashMap<>());
```

Comment:
Waarom niet gewoon: `new ConcurrentHashMap()` ?

Reply:
De ConcurrentHashMap heeft het nadeel dat het toch toelaat dat een update op hetzelfde moment kan voorkomen als een read, doordat read-operations niet blocking zijn. Denk je dat we die zekerheid willen hebben of liever de reads zo snel mogelijk maken?

Reply:
Ik zou focussen op dat de reads zo snel mogelijk gebeuren, wat cache-misses in het begin lijkt me niet zo erg.